### ISSUE TITLE
Not getting service maps 

### ISSUE THREAD

**Issue by rlex** [coroot/coroot#3]
Coroot 0.2.2
Coroot-node-agent 1.0.19
k3s [v1.24.4+k3s1](https://github.com/k3s-io/k3s/releases/tag/v1.24.4%2Bk3s1)

Status reports everything is ok:
```
prometheus: ok
coroot-node-agent: 7 nodes found
kube-state-metrics: 186 applications found
```

As an example, i took loki in distributed setup. 

In app search, i can see them: 
![image](https://user-images.githubusercontent.com/84462/191134822-a52a546d-0dc9-4162-98cf-b3b6d877e54d.png)

But if i open app details, i can't see inter-communication between components:
![image](https://user-images.githubusercontent.com/84462/191135052-e5d465c2-4781-4760-b0d9-6f1ee7f5442c.png)

Wondering if i'm missing something? Maybe specific labels?

**Comment by def** [comment]
Hey @rlex!

Is your Loki under the load? I mean, there must be active connections between components to display them on the service map.
Can you please check the following metrics in your Prometheus? 
* `rate(container_net_tcp_successful_connects_total{container_id=~".+loki-distributor.+"}[1m])`
* `container_net_tcp_active_connections{container_id=~".+loki-distributor.+”}`

To reduce the number of apps displayed on the overview page, we intentionally hide apps that have no active inbound or outbound connections. Otherwise, many auxiliary services such as sshd, udevd, etc. would be shown there.

**Comment by rlex** [comment]
Yes, there is constant traffic flow. Some parts are getting picked up, but in a strange way:

<img width="1118" alt="Screenshot 2022-09-20 at 21 13 09" src="https://user-images.githubusercontent.com/84462/191334025-11e9b97b-eecb-4a4d-a8da-cc5f4c9d930d.png">

Interesting that loki-index-gateway is designated as "external endpoint" while it's part of k8s service-cidr.

As for metrics, they are empty. Only two pods appears under .*loki.* regex

**Comment by rlex** [comment]
I deployed it on usual kubernetes and looks like everything is working. But it also have calico as a cni with kube-proxy.

In my setup, it's cilium with ebpf dataplane and DSR enabled. DSR shouldn't affect internal cluster networking, though.

**Comment by def** [comment]
It looks like the issue is related to k3s, not CNI. Is there any way to reproduce your k3s+Loki setup? I'd look into it.

**Comment by rlex** [comment]
It's fairly typical setup, but i deploy coredns and CNI manually. I also deploy hetzner cloud-controller. 

I deploy it with [my ansible role](https://github.com/rlex/ansible-role-k3s)

**Comment by def** [comment]
@rlex , it seems like we fixed this issue. 
Could you check this by running coroot:0.3.0 + coroot-node-agent:1.0.20 on your k3s cluster?

**Comment by rlex** [comment]
Sadly no, no luck. k3s is now consided part of control-plane which makes map less cluttered, but there is still no proper links between majority of services, and service-cidr is still considered "external endpoint".
So in my case:
1) pod CIDR is 10.121.0.0/24
2) service CIDR is 10.43.0.0/16

And sometimes both IPs in pod cidr and service cidr is listed as "external endpoint"

But only links i saw in several days are those: 

<img width="1236" alt="Screenshot 2022-09-22 at 20 35 03" src="https://user-images.githubusercontent.com/84462/191814414-32585088-777b-4f30-913a-b234572b5058.png">
<img width="1198" alt="Screenshot 2022-09-22 at 20 35 24" src="https://user-images.githubusercontent.com/84462/191814423-d6ba5d38-aab0-4bd0-aaf5-28608992211e.png">

If you have some VM for tests i can try to create almost 1:1 config for debugging.

**Comment by def** [comment]
I reproduced the problem on a k3s+cilium cluster. The bug is that node-agent expected all connections to be present in the conntrack table. In the case of Cilium, the destination of a TCP connection has been overwritten at the eBPF level, bypassing conntrack.
This is fixed in coroot-node-agent 1.0.21

**Comment by rlex** [comment]
Can confirm it works now. Thanks alot for great product and great communication!

**Comment by def** [comment]
Thank you for the detailed bug report!
